File: DTOs/ProductPricingDto.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace cms_webapi.DTOs
{
    public class ProductPricingDto
    {
        public long Id { get; set; }
        public string ErpProductCode { get; set; } = string.Empty;
        public string ErpGroupCode { get; set; } = string.Empty;
        public decimal ListPrice { get; set; }
        public decimal CostPrice { get; set; }
        public decimal? Discount1 { get; set; }
        public decimal? Discount2 { get; set; }
        public decimal? Discount3 { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime? UpdatedDate { get; set; }
        public DateTime? DeletedDate { get; set; }
        public bool IsDeleted { get; set; }
    }

    public class CreateProductPricingDto
    {
        [Required]
        [MaxLength(50)]
        public string ErpProductCode { get; set; } = string.Empty;

        [Required]
        [MaxLength(50)]
        public string ErpGroupCode { get; set; } = string.Empty;

        [Required]
        public decimal ListPrice { get; set; }

        [Required]
        public decimal CostPrice { get; set; }

        public decimal? Discount1 { get; set; }

        public decimal? Discount2 { get; set; }

        public decimal? Discount3 { get; set; }
    }

    public class UpdateProductPricingDto
    {
        [Required]
        [MaxLength(50)]
        public string ErpProductCode { get; set; } = string.Empty;

        [Required]
        [MaxLength(50)]
        public string ErpGroupCode { get; set; } = string.Empty;

        [Required]
        public decimal ListPrice { get; set; }

        [Required]
        public decimal CostPrice { get; set; }

        public decimal? Discount1 { get; set; }

        public decimal? Discount2 { get; set; }

        public decimal? Discount3 { get; set; }
    }
}

File: WebApi/Models/ProductPricingDto.ts
export interface ProductPricingDto {
  id: number;
  erpProductCode: string;
  erpGroupCode: string;
  listPrice: number;
  costPrice: number;
  discount1?: number;
  discount2?: number;
  discount3?: number;
  createdDate: string;
  updatedDate?: string;
  deletedDate?: string;
  isDeleted: boolean;
}

export interface CreateProductPricingDto {
  erpProductCode: string;
  erpGroupCode: string;
  listPrice: number;
  costPrice: number;
  discount1?: number;
  discount2?: number;
  discount3?: number;
}

export interface UpdateProductPricingDto {
  erpProductCode: string;
  erpGroupCode: string;
  listPrice: number;
  costPrice: number;
  discount1?: number;
  discount2?: number;
  discount3?: number;
}